### PR TITLE
[DPEDE-1978] Fix search input autocomplete active item

### PR DESCRIPTION
**Pull request by emerson-pereira**
https://ctl.atlassian.net/browse/DPEDE-1978

### CODE REVIEW

File: src/custom-elements/src/components/search-input/search-input.tsx [hunk 1]
```diff
@@ -165,6 +165,14 @@ export class SearchInput {
     const newValue = (valueChange.target as HTMLInputElement).value;
     const isAutocomplete = this._isAutocomplete();
 
+    if (!newValue) {
+      this.selectedItem = undefined;
+
+      const dropdown = this._getAutocompleteDropdown();
+
+      dropdown.resetSelection();
```
Comment (ASanchLumen) [inline src/custom-elements/src/components/search-input/search-input.tsx:173]: For reusability, we can extract this to a function (e.g. `resetSelection`) and use it also in lines 272-280

Comment (emerson-pereira) [reply src/custom-elements/src/components/search-input/search-input.tsx]: Good one, done

**Comment by emerson-pereira** [comment]
@wojciechwlodarczyklumen I confirmed with @claradamcho and the dropdown should retain selection at type and reset selection at type with empty search.

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 2]
```diff
@@ -344,6 +348,7 @@ export class Dropdown {
   onMenuItemsChange() {
     this.setMenuHeight();
     this._addClickEventListeners();
+
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:351]:
```suggestion
```

File: src/custom-elements/src/components/dropdown/dropdown.tsx [hunk 3]
```diff
@@ -494,6 +502,22 @@ export class Dropdown {
...
   }
 
+  /**
+   * Resets selection
+   */
+  @Method()
+  async resetSelection() {
+    this._value = [];
+
+    const menuItems = this._getDropdownMenuItems();
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/dropdown/dropdown.tsx:512]:
```suggestion
    const menuItems = this._getDropdownMenuItems();
    
    this._value = [];
```

File: src/custom-elements/src/components/search-input/search-input.tsx [hunk 4]
```diff
@@ -266,6 +270,16 @@ export class SearchInput {
...
+
+    this._resetSelection();
+  }
+
+  _resetSelection() {
+    this.selectedItem = undefined;
+
+    const dropdown = this._getAutocompleteDropdown();
+
+    dropdown.resetSelection();
```
Comment (josecarlosgonzalezv) [inline src/custom-elements/src/components/search-input/search-input.tsx:282]:
```suggestion
    const dropdown = this._getAutocompleteDropdown();

    this.selectedItem = undefined;
    dropdown.resetSelection();
```

**Review by josecarlosgonzalezv — APPROVED** [review]

**Review by ASanchLumen — APPROVED** [review]

**Review by jllr — APPROVED** [review]

**Pull request merged**